compiler/resolution.common/src/org/jetbrains/kotlin/resolve/calls/tower/CandidateApplicability.kt

```diff
@@ -13,7 +13,8 @@ enum class CandidateApplicability {
     INAPPLICABLE, // arguments have wrong types
     IMPOSSIBLE_TO_GENERATE, // access to outer class from nested
     RUNTIME_ERROR, // problems with visibility
-    MAY_THROW_RUNTIME_ERROR, // unsafe call or unstable smart cast
+    UNSAFE_CALL, // receiver nullability doesn't match
```

Comment:
The order of the applicability does matter: https://github.com/JetBrains/kotlin/blob/41b395b0aaa8a4404dc37c523d8e28966b531738/compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/calls/CandidateCollector.kt#L35

So maybe you can add this next to INAPPLICABLE_WRONG_RECEIVER

Reply:
Looking at the test case, it seems having it here is more appropriate. For example https://www.diffchecker.com/6IzXnZMo. On the left is the current code, on the right is moving `UNSAFE_CALL` up near `INAPPLICABLE_WRONG_RECEIVER`

Reply:
Hmm that does look better, even better than the FE1.0 error.